File: movieApp/app/src/main/java/com/lbc/practice/movieapp/view/main/MainPresenter.java
package com.lbc.practice.movieapp.view.main;

import android.util.Log;

import com.lbc.practice.movieapp.adapter.PlayingListContract;
import com.lbc.practice.movieapp.callback.OnItemClickListener;
import com.lbc.practice.movieapp.model.PlayingMovieResult;
import com.lbc.practice.movieapp.model.resource.MovieDataSource;
import com.lbc.practice.movieapp.model.resource.remote.MovieRemoteDataSource;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

public class MainPresenter implements MainContract.MainPre {

    MovieDataSource movieDataSource;
    MainContract.MainView mainView;
    PlayingListContract.View adapterView;
    PlayingListContract.PlayingModel adapterModel;
    List<PlayingMovieResult.ResultsBean> resultsBeans;


    public MainPresenter(MovieDataSource movieDataSource,  MainContract.MainView mainView) {
        this.movieDataSource = movieDataSource;
        this.mainView = mainView;
        resultsBeans = new ArrayList<PlayingMovieResult.ResultsBean>();

    }

    @Override
    public void movieInfo(String apiKey,String language, int page) {
        movieDataSource.seeMovieInfo(apiKey, language, page, new MovieDataSource.LoadDataCallBack() {
            @Override
            public void onLoadData(List<PlayingMovieResult.ResultsBean> movieInfo) {
                Log.e("retropre",movieInfo.get(0).getTitle());

                    resultsBeans.addAll(movieInfo);
                    adapterModel.addMovieInfo(resultsBeans);
                    adapterView.notifyAdapter();
                    mainView.increasePageCount();
            }

            @Override
            public void onFailData(String errorMsg) {
                    mainView.errorMassage(errorMsg);
            }
        });
    }



    @Override
    public void setPlayingAdaterModel(PlayingListContract.PlayingModel model) {
        this.adapterModel = model;
    }

    @Override
    public void setPlayingAdaterView(PlayingListContract.View view) {
        this.adapterView =view;
        adapterView.setOnclickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(int position) {
                mainView.moveToSecond(resultsBeans.get(position));
            }
        });
    }


}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/view/main/MainPresenter.kt
package com.lbc.practice.movieapp.view.main;

import android.util.Log;

import com.lbc.practice.movieapp.adapter.PlayingListContract;
import com.lbc.practice.movieapp.callback.OnItemClickListener
import com.lbc.practice.movieapp.data.PlayingMovieResult;
import com.lbc.practice.movieapp.data.resource.MovieDataSource;
import com.lbc.practice.movieapp.data.resource.MovieRepository

import java.util.ArrayList;
import javax.inject.Inject

class MainPresenter : MainContract.MainPre {


    var adapterView: PlayingListContract.View?=null
    var adapterModel: PlayingListContract.PlayingModel?=null
    var resultsBeans: MutableList<PlayingMovieResult.ResultsBean>?=null
    var mainView : MainContract.MainView ?= null
    var movieDataSource: MovieDataSource

    @Inject
    constructor(movieRepository : MovieRepository) {
        movieDataSource =movieRepository
    }

    init {
        resultsBeans = ArrayList()
    }

    override fun takeView(mainView: MainContract.MainView) {
        this.mainView = mainView
    }

    override fun movieInfo(apiKey: String, language: String, page: Int) {
        movieDataSource.seeMovieInfo(apiKey, language, page, object : MovieDataSource.LoadDataCallBack {
            override fun onLoadData(movieInfo: kotlin.collections.List<PlayingMovieResult.ResultsBean>) {
                Log.e("retropre", movieInfo[0].title)

                resultsBeans?.addAll(movieInfo)
                adapterModel?.addMovieInfo(resultsBeans!!)
                adapterView?.notifyAdapter()
                mainView!!.increasePageCount()
            }

            override fun onFailData(errorMsg: String) {
                mainView!!.errorMassage(errorMsg)
            }
        })
    }


    override fun setPlayingAdaterModel(model: PlayingListContract.PlayingModel) {
        this.adapterModel = model
    }

    override fun setPlayingAdaterView(view: PlayingListContract.View) {
        this.adapterView = view
        adapterView?.setOnclickListener(object : OnItemClickListener {
            override fun onItemClick(position: Int) {
                mainView!!.moveToSecond(resultsBeans!!.get(position))
            }
        })

//        adapterView!!.setOnclickListener(position-> {
//            mainView.moveToSecond(resultsBeans!![position])
//        })
    }


}
